### PR TITLE
Unit tests for PrimaryKey field type change

### PR DESCRIPTION
**Pull request by stk1m1**
Unit tests for checking proper migration of PrimaryKey field type change from String to Integer/int. These cases were surfaced in issue #3068.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,11 @@
 
 ### Bug fixes
 
-* Disabled the optional API transformer since it has problems with DexGuard (3022).
+* Disabled the optional API transformer since it has problems with DexGuard (#3022).
+
+### Enhancements
+
+* Unit tests for checking proper migration of primary key field type change from `String` to `Integer`/`int` (#3068).
```
Comment (cmelchior) [inline CHANGELOG.md]: We normally don't put internal changes in the changelog

**Comment by stk1m1** [comment]
@realm/java please review.

File: realm/realm-library/src/androidTest/java/io/realm/entities/migration/MigrationFieldTypeToInt.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+// Class used to test what happens if you modify a String type primary key field to int type in a migration.
+public class MigrationFieldTypeToInt extends RealmObject implements MigrationPrimaryKey {
+    public static String CLASS_NAME = "MigrationFieldTypeToInt";
+
+    public static long DEFAULT_FIELDS_COUNT = 5;
+    public static long DEFAULT_PRIMARY_INDEX = 4;
+
+    public static String FIELD_PRIMARY = "fieldIntPrimary";
+
+    private Byte fieldFirst;
```
Comment (beeender) [inline realm/realm-library/src/androidTest/java/io/realm/entities/migration/MigrationFieldTypeToInt.java]: The test class can be much shorter by removing the getters/setters and just using the public field.

**Comment by beeender** [comment]
:+1: with a minor comment

File: realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java [hunk 3]
```diff
@@ -374,6 +376,126 @@ public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
         assertEquals(MigrationFieldRenamed.FIELD_PRIMARY, objectSchema.getPrimaryKey());
     }
 
+    private void createObjectsWithOldPrimaryKey(final String className, final boolean insertNullValue) {
+        DynamicRealm realm = DynamicRealm.getInstance(configFactory.createConfigurationBuilder().build());
+        realm.executeTransaction(new DynamicRealm.Transaction() {
```
Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java]: Let's use `try-finally`

**Comment by zaki50** [comment]
I don't think you need to update `asset_file.realm` since I recently added a module class for that Realm file.

**Comment by stk1m1** [comment]
retest this please

**Comment by zaki50** [comment]
:+1:

**Comment by kneth** [comment]
:+1:

**Pull request merged**